src/generic/stage2/numberparsing.h

```diff
@@ -179,26 +167,12 @@ simdjson_really_inline bool compute_float_64(int64_t power, uint64_t i, bool neg
   // which we guard against.
   // If we have lots of trailing zeros, we may fall right between two
   // floating-point values.
-  if (simdjson_unlikely((lower == 0) && ((upper & 0x1FF) == 0) &&
+  // We have 5**27 < 2**64 and it is the largest power of 5 to do so.
+  if (simdjson_unlikely((lower == 0) && (power >= -27) && (power <= 27)  &&
                ((mantissa & 3) == 1))) {
-    // if mantissa & 1 == 1 we might need to round up.
-    //
-    // Scenarios:
-    // 1. We are not in the middle. Then we should round up.
-    //
-    // 2. We are right in the middle. Whether we round up depends
-    // on the last significant bit: if it is "one" then we round
-    // up (round to even) otherwise, we do not.
-    //
-    // So if the last significant bit is 1, we can safely round up.
-    // Hence we only need to bail out if (mantissa & 3) == 1.
-    // Otherwise we may need more accuracy or analysis to determine whether
-    // we are exactly between two floating-point numbers.
-    // It can be triggered with 1e23.
-    // Note: because the factor_mantissa and factor_mantissa_low are
-    // almost always rounded down (except for small positive powers),
-    // almost always should round up.
-    return false;
+    if((mantissa  << (upperbit + 64 - 53 - 2)) ==  upper) {
+      mantissa ^= 1;             // flip it so that we do not round up
```

Comment:
I think this would be clearer as mantissa &= ~1 ... using ^ implies we think it might be 0 or 1 to start with, but we know it's 1.

Reply:
Done.